python/taichi/misc/np2ply.py

```diff
@@ -99,20 +99,17 @@ def add_vertex_normal(self, normal):
         self.add_vertex_channel("ny", "float", normal[:, 1])
         self.add_vertex_channel("nz", "float", normal[:, 2])
 
-    def add_vertex_color(self, r: np.array, g: np.array, b: np.array):
+    def add_vertex_rgb(self, r: np.array, g: np.array, b: np.array):

```

Comment:
Can we keep a 'add_vertex_color' method in this class? It can be calling add_vertex_rgb directly, but it's really intuitive to use. Thanks

Reply:
Sure, what about this way, this will be easier for user to distinct the difference input list. But I can also do it the old way (using overload).
```
def add_vertex_rgb(self, r, g, b)

def add_vertex_color(self, color: np.ndarray/ti.vector)
```

Similarly we can define `add_vertex_xyz` and `add_vertex_pos`.

Reply:
I think from ti to np should be fairly easy so it's should be alright if we stick everything with np. Overloading sounds great but can it be done gracefully in python?

Reply:
Haha, conversion between ti and np was not my point. I did not well illustrate the term 'overloading' previously, its actually because in the current commit I have both `add_vertex_pos(x,y,z)` and `add_vertex_pos(pos: vector)` which is the overload. Though it works well, it can be confusing for people and pylint.

So following your previous suggestion about keeping both `add_vertex_color` and `add_vertex_rgb`, I was suggesting we can define them as below to both avoid overloading and distinguish the input list, meanwhile keeping both.

```
def add_vertex_rgb(self, r, g, b)

def add_vertex_color(self, colorVector)

def add_vertex_xyz(self, x, y, z)

def add_vertex_pos(self, posVector)
```

Sorry for the confusion :>

Reply:
To be honest I still prefer only have setters for pos, color and rgba, and have all channels individually passed in. It's the least confusing way, I would suggest to keep it simple and less over designed 😁

Reply:
We both got a point and timeline is urgent, we still have tutorials to build, so for now I will try to do it my fashion and try to make it clear on the documentation
@__@ T__T although I have applied many suggestions, it makes me feel a little down to not accept all of other's kind suggestions. But time is limited, I will leave this conversation open if we got feedback from the students later.

Reply:
Well, no offenses here but the fact is that no one wills to remember what is the difference between color and rgb, pos and xyz. **It was not designed that way and shouldn't be like that either**. We have to consider the user when we do something like this. You can say, ok for the first version let's cut the corners and leave some confusions that we promise to fix when we got feedback. However, when we fix that, there'll be more **silent dissatisfaction** from the users that have got used to our tangled logic. Therefore it's really important to have a design before having it done, and that's is what I did last weekend. I agree that we could make this feature more convenient by all your suggestions and revisions, but sorry that's exactly why I tried to avoid in my code - forget about convenience in the first version, make it robust, **do whatever it means to do without discouragement**. Convenience can be compromised and delayed to the future, but not the above qualities.

Reply:
Got your point, I will adopt your suggestion concerning the user design issue and comment out the compact/reloaded ones.

Thanks for all the patience :>